### PR TITLE
Initialising start and end for range equal to nil condition

### PR DESCRIPTION
**Pull request by Tulsishah**

### CODE REVIEW

File: internal/storage/bucket_handle.go [hunk 1]
```diff
@@ -41,8 +41,12 @@ func (bh *bucketHandle) NewReader(
 	ctx context.Context,
 	req *gcs.ReadObjectRequest) (rc io.ReadCloser, err error) {
 	// Initialising the starting offset and the length to be read by the reader.
-	start := int64((*req.Range).Start)
-	end := int64((*req.Range).Limit)
+	start := int64(0)
+	end := int64(-1)
```
Comment (vadlakondaswetha) [inline internal/storage/bucket_handle.go:45]: setting end to -1 and computing length from it is confusing. can you directly set the length here and update it in the if block below like this:

length: -1
if(rangle !=nil) {
end = int64(req.Range.limit)
length = end-start
}

File: internal/storage/bucket_handle.go [hunk 2]
```diff
@@ -41,8 +41,12 @@ func (bh *bucketHandle) NewReader(
 	ctx context.Context,
 	req *gcs.ReadObjectRequest) (rc io.ReadCloser, err error) {
 	// Initialising the starting offset and the length to be read by the reader.
-	start := int64((*req.Range).Start)
-	end := int64((*req.Range).Limit)
+	start := int64(0)
```
Comment (vadlakondaswetha) [inline internal/storage/bucket_handle.go:44]: do we need to make it int64 explicitly?

File: internal/storage/bucket_handle.go [hunk 3]
```diff
@@ -41,8 +41,12 @@ func (bh *bucketHandle) NewReader(
...
+	end := int64(-1)
+	if req.Range != nil {
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply internal/storage/bucket_handle.go:48]: fixed

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: internal/storage/bucket_handle.go (hunk 2 above)
Comment (Tulsishah) [reply internal/storage/bucket_handle.go:44]: yes because req.range is returning in uint64 but in reader method the parameter type is int64.

File: internal/storage/bucket_handle.go (hunk 1 above)
Comment (Tulsishah) [reply internal/storage/bucket_handle.go]: fixed

File: internal/storage/bucket_handle.go [hunk 4]
```diff
@@ -41,9 +41,15 @@ func (bh *bucketHandle) NewReader(
 	ctx context.Context,
 	req *gcs.ReadObjectRequest) (rc io.ReadCloser, err error) {
 	// Initialising the starting offset and the length to be read by the reader.
-	start := int64((*req.Range).Start)
-	end := int64((*req.Range).Limit)
-	length := int64(end - start)
+	start := int64(0)
+	length := int64(-1)
+	// If Range equals to nil, it has a default call which has start and length as above values otherwise it will take mentioned range.
```
Comment (vadlakondaswetha) [inline internal/storage/bucket_handle.go:46]: rephrase it to  Following the semantics of NewReader method. Passing start, length as 0,-1 reads the entire file.

**Review by vadlakondaswetha — APPROVED** [review]
